java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -75,11 +77,50 @@ public JObject(
     } else {
       String nextPackagePath = pkgPrefix + AbstractJSONSchema2Pojo.packageName(this.className);
 
+      // in order to handle duplicated fields, first let's build a map of fields grouped by their sanitized names, i.e.:
+      // 1(fieldName) -> n(fieldDefinition(key, props))
+      final Map<String, Map<String, JSONSchemaProps>> groupedFieldDefinitions = fields.entrySet().stream()
+          .collect(Collectors.groupingBy(
+              (f) -> AbstractJSONSchema2Pojo.sanitizeString(f.getKey()),
+              Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
+        String fieldKey = field.getKey();
+        // lookup the duplicated field properties map
+        final Map<String, JSONSchemaProps> fieldDuplicatesDefinition = groupedFieldDefinitions
+            .get(AbstractJSONSchema2Pojo.sanitizeString(field.getKey()));
+        final int duplicatesCount = fieldDuplicatesDefinition.entrySet().size();
+        if (duplicatesCount > 1) {
+          // ok, duplicates exist...
+          // we want to throw an exception on some duplicates missing requirements. the first one that we enforce is
+          // for exactly 1 field duplicate to exist
+          if (duplicatesCount > 2) {
+            throw new JavaGeneratorException(
+                String.format("The %s field has %d duplicates, which is not a supported configuration",
+                    field.getKey(),
+                    duplicatesCount - 1));
+          }
+          // another requirement that we enforce is that if one field duplicate exists, then it's because it has
+          // been marked as deprecated
+          final long deprecatedDuplicates = fieldDuplicatesDefinition.entrySet().stream()
```

Comment:
I think we can use `anyMatch` instead of counting exactly the number of duplicates (now it should be a fixed number).

Reply:
I'd use `noneMatch`:

```
final Boolean noDeprecatedDuplicatesExist = fieldDuplicatesDefinition.entrySet().stream()
              .noneMatch(d -> d.getValue().getDescription().trim().toLowerCase().startsWith(DEPRECATED_FIELD_MARKER));
```